docs/examples/couchbase.md

```diff
@@ -0,0 +1,272 @@
+
+### Setup credentials for the instance
+
+```
+curl -v -X POST http://192.168.99.100:8091/settings/web -d port=8091 -d username=Administrator -d password=password
+* Hostname was NOT found in DNS cache
+*   Trying 192.168.99.100...
+* Connected to 192.168.99.100 (192.168.99.100) port 8091 (#0)
+> POST /settings/web HTTP/1.1
+> User-Agent: curl/7.37.1
+> Host: 192.168.99.100:8091
+> Accept: */*
+> Content-Length: 50
+> Content-Type: application/x-www-form-urlencoded
+> 
+* upload completely sent off: 50 out of 50 bytes
+< HTTP/1.1 200 OK
+* Server Couchbase Server is not blacklisted
+< Server: Couchbase Server
+< Pragma: no-cache
+< Date: Wed, 25 Nov 2015 22:50:43 GMT
+< Content-Type: application/json
+< Content-Length: 44
+< Cache-Control: no-cache
+< 
+* Connection #0 to host 192.168.99.100 left intact
+{"newBaseUri":"http://192.168.99.100:8091/"}
+```
+
+### Install sample data
+
+Load some sample data in the Couchbase instance.
+
+```
+curl -v -u Administrator:password -X POST http://192.168.99.100:8091/sampleBuckets/install -d '["travel-sample"]'
+* Hostname was NOT found in DNS cache
+*   Trying 192.168.99.100...
+* Connected to 192.168.99.100 (192.168.99.100) port 8091 (#0)
+* Server auth using Basic with user 'Administrator'
+> POST /sampleBuckets/install HTTP/1.1
+> Authorization: Basic QWRtaW5pc3RyYXRvcjpwYXNzd29yZA==
+> User-Agent: curl/7.37.1
+> Host: 192.168.99.100:8091
+> Accept: */*
+> Content-Length: 17
+> Content-Type: application/x-www-form-urlencoded
+> 
+* upload completely sent off: 17 out of 17 bytes
+< HTTP/1.1 202 Accepted
+* Server Couchbase Server is not blacklisted
+< Server: Couchbase Server
+< Pragma: no-cache
+< Date: Wed, 25 Nov 2015 22:51:51 GMT
+< Content-Type: application/json
+< Content-Length: 2
+< Cache-Control: no-cache
+< 
+* Connection #0 to host 192.168.99.100 left intact
+[]
+```
+
+Congratulations, you are now running a Couchbase container, fully configured using REST API.
+
+## Query Couchbase using CBQ
+
+[CBQ](http://developer.couchbase.com/documentation/server/4.1/cli/cbq-tool.html), short for Couchbase Query, is a CLI tool that allows to query a Couchbase server. This tool is installed as part of the Couchbase Docker image.
+
+List container id of Couchbase server:
+
+```
+docker ps
+CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                                                                                               NAMES
+e54a9849ba35        couchbase/server    "/entrypoint.sh couch"   3 minutes ago       Up 3 minutes        0.0.0.0:8091-8093->8091-8093/tcp, 11207/tcp, 11211/tcp, 18091-18092/tcp, 0.0.0.0:11210->11210/tcp   couchbaseserver_mycouchbase_1
+```
+
+This output shows the complete information about the container. Alternatively, just the container id can be obtained:
+
+```
+docker ps | grep couch | awk '{print $1}'
+e54a9849ba35
+```
+
+Run Couchbase Query CLI tool:
+
+```
+docker exec -it e5 /opt/couchbase/bin/cbq
```

Comment:
Yes, I understand; point is that `docker exec` is basically just for debugging purposes. In general, we suggest people to don't use it for other purposes.

so, e.g.

```
# start the server
docker run -d --name db couchbase/server

# to start an interactive CLI session
docker run -it --rm --link db:db couchbase/server /opt/couchbase/bin/cbq
```

Ideally, `/opt/couchbase/bin/cbq` should be in the `PATH` inside the container, so that I can just do;

    docker run -it --rm --link db:db couchbase/server cbq

not sure if `cbq` needs a host specified to connect to? In this case the host would be `db` (from the link)